### PR TITLE
[ONNX] Export var, var_mean and std_mean ops

### PR DESCRIPTION
**Pull request by neginraoof**
Adding export for var, var_mean and std_mean ops

### CODE REVIEW

File: test/onnx/expect/TestOperators.test_std.expect [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ir_version: 6
 producer_name: "pytorch"
-producer_version: "CURRENT_VERSION"
+producer_versionproducer_version: "CURRENT_VERSION"
```
Comment (KsenijaS) [inline test/onnx/expect/TestOperators.test_std.expect:3]: this doesn't look right: producer_versionproducer_version -> producer_version

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2240,9 +2239,46 @@ def _std(g, input, dim, unbiased, keepdim):
 # torch.std(input, dim, keepdim=False, unbiased=True)
 def std(g, input, *args):
     if len(args) == 3:
-        return _std(g, input, *args)
+        var, _ = _var(g, input, *args)
```
Comment (KsenijaS) [inline torch/onnx/symbolic_opset9.py:2242]: could this be simplified as return out, _ = var(g, input, *args); return g.op("Sqrt", out)

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -2240,9 +2239,46 @@ def _std(g, input, dim, unbiased, keepdim):
...
+    else:
+        var, _ = _var(g, input, None, args[0], None)
+    return g.op("Sqrt", var)
+
+
+# Since position of optional arguments can change for std, this is a hack to find if first argument
+# is 'dim' or 'unbiased'. As shown below, 'dim' argument could be listed before 'unbiased' :
+# torch.var(input, unbiased=True)
+# torch.var(input, dim, keepdim=False, unbiased=True)
+def var(g, input, *args):
```
Comment (KsenijaS) [inline torch/onnx/symbolic_opset9.py:2252]: could this be simplifed as return var_mean(g, input, *args)[0]

File: torch/onnx/symbolic_opset9.py [hunk 4]
```diff
@@ -2210,28 +2210,27 @@ def gather(g, self, dim, index, sparse_grad=False):
...
-        std = g.op("Sqrt", var)
-        return std
+def _var(g, input, dim, unbiased, keepdim):
+    sqrd = g.op("Mul", input, input)
+    if dim is None:
+        sqrdmean = g.op("ReduceMean", sqrd, keepdims_i=0)
+        mean = g.op("ReduceMean", input, keepdims_i=0)
+        num_elements = numel(g, input)
     else:
-        _unimplemented("std", "Unknown input rank. Cannot compute std along dimensions.")
```
Comment (KsenijaS) [inline torch/onnx/symbolic_opset9.py:2234]: is there a case where std cannot be exported?

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:2242]: Yeah I think this is more about style than code optimization.
Both styles have the same complexity.

File: torch/onnx/symbolic_opset9.py (hunk 3 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:2252]: Thanks. Yeah I think this is more about style than code optimization.
Both styles have the same complexity.

File: torch/onnx/symbolic_opset9.py (hunk 4 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:2234]: Previously this was relying on static shape, so we couldn't export if shape was not provided. I updated this to use ONNX Shape.

File: test/onnx/expect/TestOperators.test_std.expect (hunk 1 above)
Comment (neginraoof) [reply test/onnx/expect/TestOperators.test_std.expect:3]: Thanks. Yeah this is auto generated. Will fix.

File: torch/onnx/symbolic_opset9.py [hunk 5]
```diff
@@ -2240,9 +2239,46 @@ def _std(g, input, dim, unbiased, keepdim):
...
+    return g.op("Sqrt", var)
+
+
+# Since position of optional arguments can change for std, this is a hack to find if first argument
+# is 'dim' or 'unbiased'. As shown below, 'dim' argument could be listed before 'unbiased' :
+# torch.var(input, unbiased=True)
+# torch.var(input, dim, keepdim=False, unbiased=True)
+def var(g, input, *args):
+    if len(args) == 3:
+        var, _ = _var(g, input, *args)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2254]: The above signature for `_var` is `def _var(g, input, dim, unbiased, keepdim):` and comment shows this method as possibly `torch.var(input, dim, keepdim=False, unbiased=True)`. Is `unbiased` and `keepdim` swapped?

File: torch/onnx/symbolic_opset9.py [hunk 6]
```diff
@@ -2240,9 +2239,46 @@ def _std(g, input, dim, unbiased, keepdim):
...
+    else:
+        var, _ = _var(g, input, None, args[0], None)
+    return var
+
+
+# Since position of optional arguments can change for std, this is a hack to find if first argument
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2260]:
```suggestion
# Since position of optional arguments can change for var_mean, this is a hack to find if first argument
```
likewise for the other comments.

Question for the optional values in the comment, `... keepdim=False, unbiased=True ...` They are only optional at PyTorch code level correct? Are we guaranteed to receive either among
```
# torch.var_mean(input, unbiased)
# torch.var_mean(input, dim, keepdim, unbiased)
```

Otherwise it is possible someone actually calls with `torch.var_mean(input, dim)`, and `dim` got considered as `unbiased`

**Comment by BowenBao** [comment]
CI test failure seems related

File: torch/onnx/symbolic_opset9.py (hunk 5 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:2254]: The comment should be corrected. Thanks.

File: torch/onnx/symbolic_opset9.py (hunk 6 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Yeah these are the two schemas. I'll fix the comments to be more clear. Thanks.

File: torch/onnx/symbolic_opset9.py [hunk 7]
```diff
@@ -2210,39 +2210,76 @@ def gather(g, self, dim, index, sparse_grad=False):
...
     else:
-        _unimplemented("std", "Unknown input rank. Cannot compute std along dimensions.")
+        sqrdmean = g.op("ReduceMean", sqrd, axes_i=dim, keepdims_i=keepdim)
+        mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=keepdim)
+        redudced_dims = g.op("Shape", input)
+        # dim could contain one or multiple dimensions
+        redudced_dims = g.op("Gather", redudced_dims, g.op("Constant", value_t=torch.tensor(dim)), axis_i=0)
+        num_elements = g.op("ReduceProd", redudced_dims, keepdims_i=0)
+    meansqrd = g.op("Mul", mean, mean)
+    var = g.op("Abs", g.op("Sub", sqrdmean, meansqrd))
```
Comment (spandantiwari) [inline torch/onnx/symbolic_opset9.py:2227]: Is `Abs` op required here?

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Review by bzinodev — APPROVED** [review]
